lib/service/servicecfg/database.go

```diff
@@ -100,53 +93,22 @@ type OracleOptions struct {
 
 // CheckAndSetDefaults validates the database proxy configuration.
 func (d *Database) CheckAndSetDefaults() error {
-	if err := enterprise.ProtocolValidation(d.Protocol); err != nil {
-		return trace.Wrap(err)
-	}
-	if d.Name == "" {
-		return trace.BadParameter("empty database name")
-	}
-
 	// Mark the database as coming from the static configuration.
 	if d.StaticLabels == nil {
 		d.StaticLabels = make(map[string]string)
 	}
 	d.StaticLabels[types.OriginLabel] = types.OriginConfigFile
 
-	if err := d.TLS.Mode.CheckAndSetDefaults(); err != nil {
-		return trace.Wrap(err)
-	}
-
-	// We support Azure AD authentication and Kerberos auth with AD for SQL
-	// Server. The first method doesn't require additional configuration since
-	// it assumes the environment’s Azure credentials
-	// (https://learn.microsoft.com/en-us/azure/developer/go/azure-sdk-authentication).
-	// The second method requires additional information, validated by
-	// DatabaseAD.
-	if d.Protocol == defaults.ProtocolSQLServer &&
-		(d.AD.Domain != "" || !strings.Contains(d.URI, azure.MSSQLEndpointSuffix)) {
-		if err := d.AD.CheckAndSetDefaults(d.Name); err != nil {
-			return trace.Wrap(err)
-		}
-	}
-
 	// If AWS account ID is missing, but assume role ARN is given,
 	// try to parse the role arn and set the account id to match.
 	if d.AWS.AccountID == "" && d.AWS.AssumeRoleARN != "" {
 		parsed, err := awsutils.ParseRoleARN(d.AWS.AssumeRoleARN)
-		if err != nil {
-			return trace.BadParameter("database %q invalid AWS assume_role_arn: %v",
-				d.Name, err)
+		if err == nil {
+			d.AWS.AccountID = parsed.AccountID
```

Comment:
Do you mind leaving a TODO to move this to API? We can either implement our own parser or use aws-sdk-go-v2